third_problem.rb

```diff
@@ -0,0 +1,22 @@
+class ThirdProblem
+  def initialize(numbers_array)
+    @numbers = numbers_array
+  end
+
+  def hasDuplicates
+    found = false
+    @numbers.each {|index|
+      absolute_current_index =index.abs-1
+      absolut_target_index = @numbers[absolute_current_index].abs-1
```

Comment:
You keep repeating `@numbers[absolute_current_index]`, I will think is worth extracting in a variable. Index lookup for arrays in ruby has it's cost.

Reply:
sure , extracted to variable